### DISCUSSION TITLE
How to setup extension to run tests on-demand?

### DISCUSSION THREAD

**Discussion by ollieluvic** [vitest-dev/vscode#303] [Q&A] [answered]
I have a sort-of legacy code base with problem tests that are causing Vitest to consume all available RAM and swap space since it runs all the tests on the workspace at start up. Is there a way to set it to only run tests on-demand as we have with the jest extension? I don't have the same issue when running vitest through the terminal, but I can't debug specific test causes using the terminal alone.

**Comment by sheremet-va** [comment, marked as answer]
Do you use the latest [pre-release version](https://github.com/vitest-dev/vscode/releases/tag/v0.5.6)?

**Reply by ollieluvic** [reply]
I'm currently running the stable version, but I'll try that again. In the past when I tried the pre-release it wouldn't work altogether.

**Reply by ollieluvic** [reply]
@sheremet-va Updating the project's vitest dep to 1.4.x and switching the extension to pre-release (v0.5.6) fixed the issue. Thanks for the insight.

**Comment by fregante** [comment]
It would be great to have the answer to the question in title since the accepted answer fixes an unrelated problem @sheremet-va

**Reply by sheremet-va** [reply]
What question? The extension runs tests on demand.

**Reply by fregante** [reply]
Indeed. However I see that with the extension enabled I get 4 more `node` processes just by opening VS Code, this matches the number of vitest.config files. The processes stay open for the entire session. Disabling the extension and reloading VS Code does not start them, I'm down to a single `node` process open.

So now I wonder what they're doing

**Reply by sheremet-va** [reply]
The extension starts Vitest in standalone mode, asks for test files and parses them when a test file is opened. It also reuses them when running the tests, but it doesn’t run tests on start.

**Reply by svda** [reply]
I think this is a very valid concern. I tried using this extension in a monorepo with ~200 config files. It opens a lot of terminal sessions (I need 'terminal' mode to initialize the environment properly otherwise vitest explorer doesn't work). Initialization takes ages, up until the point vscode becomes sluggish and I gave up. I would love to see a way it can handle repositories this big.

**Reply by jackbonaguro-dandy** [reply]
Same here! Monorepo with dozens of vitest configs, on a memory-constrained machine. It would be great to only run any processes on demand.
